### PR TITLE
Improve zorder demo

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

- Add zorder of more artists
- Reference `axisbelow` property
- Code cleanup

### CODE REVIEW

File: examples/misc/zorder_demo.py [hunk 1]
```diff
@@ -3,66 +3,69 @@
...
-=======================    =======
+================================================================    =======
+Artist                                                              Z-order
+================================================================    =======
+Images (`.AxesImage`, `.FigureImage`, `.BboxImage`)                 0
+`.Patch`, `.PatchCollection`                                        1
+`.Line2D`, `.LineCollection` (including minor ticks, grid lines)    2
+Major ticks                                                         2.01
+`.Text` (including axes labels and titles)                          3
+`.Legend`                                                           5
```
Comment (jklymak) [inline examples/misc/zorder_demo.py]: Many of these classes are a bit obscure to users.  Can we include the `Axes` calls that make them?

Comment (timhoffm) [reply examples/misc/zorder_demo.py]: I agree that these classes are a bit obscure. OTOH, the Z-order is exactly attached to these classes. It does not match too well to `Axes` calls, at least I found no consistent and descriptive way: `Line2D` can be created by a number of methods: `plot` (and all of its derivatives like `semilogx`, ...), `axh/vline`, `scatter` (depending on the parameters), ... Major ticks are implicitly made, there's no `Axes` method that create them. Even though `Text` and `Rectangle` are also building blocks for tables, I would not reference `table()` as a patch-creating method.

Therefore, I sticked to the existing structure of the table and just added more entries. Ideas to improve on that are always welcome.

Comment (jklymak) [reply examples/misc/zorder_demo.py]: I would just do `e.g. ax.plot, including minor...` without trying to be comprehensive...

Comment (timhoffm) [reply examples/misc/zorder_demo.py]: Feel free to push an update if you like. I'm not feeling like fiddling what commands should get added right now.

Anyway, IMHO this PR stands for for itself and could get merged as an incremental improvement as is. Further improvements are always possible later.

File: examples/misc/zorder_demo.py [hunk 2]
```diff
@@ -3,66 +3,69 @@
...
+================================================================    =======
 
-You can change the order for individual artists by setting the zorder.  Any
-individual plot() call can set a value for the zorder of that particular item.
+Any individual `~.axes.Axes.plot()` call can set a value for the zorder of that
```
Comment (anntzer) [inline examples/misc/zorder_demo.py]: really that should be "any call to a plotting method" (in general, not just plot()).

**Pull request merged**